### PR TITLE
Added a new function extract_stokes_dof_indices 

### PR DESCRIPTION
**Pull request by yinghe616**
To address issue #1640 
Added a new function extract_stokes_dof_indices for type 
internal::As…sembly::CopyData::StokesSystem

### CODE REVIEW

File: include/aspect/assembly.h [hunk 1]
```diff
@@ -250,6 +250,9 @@ namespace aspect
 
           FullMatrix<double>          local_matrix;
           std::vector<types::global_dof_index>   local_dof_indices;
+          void extract_stokes_dof_indices(const Introspection<dim> &introspection,
```
Comment (bangerth) [inline include/aspect/assembly.h]: can you add an empty line and a comment for this new function?

File: include/aspect/assembly.h [hunk 2]
```diff
@@ -250,6 +250,9 @@ namespace aspect
...
+                                          internal::Assembly::Scratch::StokesPreconditioner<dim> &scratch,
```
Comment (bangerth) [inline include/aspect/assembly.h]: you can make this second argument `const` as well.

**Comment by yinghe616** [comment]
Comments addressed.

**Comment by bangerth** [comment]
Looks good.

**Pull request merged**